src/workerd/api/sockets.c++

```diff
@@ -121,13 +121,15 @@ jsg::Ref<Socket> connectImplNoOutputLock(
           TypeError, "Specified address could not be parsed.");
       auto& host = JSG_REQUIRE_NONNULL(record.host, TypeError,
           "Specified address is missing hostname.");
-      port = JSG_REQUIRE_NONNULL(record.port, TypeError,
-          "Specified address is missing a port.");
+      // Note that there is an edge case here where the address containing `:443` will nonetheless
+      // parse to a record whose `port` is set to nullptr.
+      auto port = record.port.orDefault(443);
+      isFetchablePort = port == 443 || port == 80;
       domain = host.toStr();
     }
     KJ_CASE_ONEOF(record, SocketAddress) {
       domain = kj::heapString(record.hostname);
-      port = record.port;
+      isFetchablePort = record.port == 443 || record.port == 80;
```

Comment:
Are you sure these are the only ports, what about 8080 and 8443?

Reply:
It's only the default ports that get omitted.
![image](https://github.com/cloudflare/workerd/assets/439929/852280ec-7c41-4f5f-9b01-1899f9912cea)